vendors/espressif/boards/esp32/ports/ble/nimble/iot_ble_hal_gap.c

```diff
@@ -667,6 +671,18 @@ BTStatus_t prvBTSetAdvData( uint8_t ucAdapterIf,
         /* fixme: set adv_params->high_duty_cycle accordingly */
     }
 
+    if( pxParams->bSetScanRsp == false )
+    {
+        if( pxParams->usTimeout != 0 )
+        {
+            lAdvDurationMS = ( int32_t ) ( pxParams->usTimeout * IOT_BLE_ADVERTISING_DURATION_UNITS ); /* Multiply pxParams->usTimeout value by units = 10ms */
```

Comment:
Not understand this math computation. What is the unit here?

Reply:
This is as per BLE spec. The input integer value is multiplied by 10ms to get the actual timeout value for advertisement.

Reply:
I see.

Reply:
The comment on line 678 is redundant and will become incorrect if the value of `IOT_BLE_ADVERTISING_DURATION_UNITS` is ever removed.  I would delete it.